ports/nrf/boards/blueio_tag_evim/mpconfigboard.h

```diff
@@ -0,0 +1,77 @@
+/*
+ * This file is part of the Micro Python project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2019 I-SYST inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#define MICROPY_HW_BOARD_NAME       "BLUEIO-TAG-EVIM"
+#define MICROPY_HW_MCU_NAME         "NRF52832"
+#define MICROPY_PY_SYS_PLATFORM     "BLYST Nano"
+
+#define MICROPY_PY_MACHINE_SOFT_PWM (1)
+#define MICROPY_PY_MUSIC            (1)
+
+#define MICROPY_PY_MACHINE_UART     (1)
+#define MICROPY_PY_MACHINE_HW_PWM   (1)
+#define MICROPY_PY_MACHINE_HW_SPI   (1)
+#define MICROPY_PY_MACHINE_TIMER    (1)
+#define MICROPY_PY_MACHINE_RTCOUNTER (1)
+#define MICROPY_PY_MACHINE_I2C      (1)
+#define MICROPY_PY_MACHINE_ADC      (1)
+#define MICROPY_PY_MACHINE_TEMP     (1)
+#define MICROPY_PY_RANDOM_HW_RNG    (1)
+
+#define MICROPY_HW_HAS_LED          (1)
+#define MICROPY_HW_LED_COUNT        (4)
+#define MICROPY_HW_LED_PULLUP       (1)
+
+#define MICROPY_HW_LED1             (30) // LED1
+#define MICROPY_HW_LED1_LEVEL		(0)
```

Comment:
1. Is it correct to say that if LEVEL is 1, its either lack of PULLUP or there is a PULL resistor to GND (active high)? 

2. Would it make sense to name it LED1_PULLUP? PULLUP will align with the existing config, while LED**X**_PULLUP will indicate the specific override of the generic.

3. Would not LED2, 3, and 4 be covered by MICROPY_HW_LED_PULLUP (1)? It could be enough to only add an override on LED1?

Also, it would look nice to get the values tab-aligned :)

Reply:
@glennrub I think PULLUP (1) is same as LEVEL (0) and PULLUP(0) as LEVEL (1).  I tried not affect the old boards by adding the LEDx_LEVEL.  Because LED_PULLUP is common to all LED.  If I changed it to LEDx_PULLUP would need to change all the boards.  I think active level is more meaningful than pullup.  Because the LED can be pullup and still active high if driven by a transistor instead of directly connected to the gpio pin.  This is the case of the BLUEIO-TAG-EVIM board.  The nRF52 is powered at 1.8V. It cannot directly drive the LED.  Therefore transistor is used to drive it at 3V. 
Sorry the TAB, I used tab 4 in Eclipse.  What is the standard TAB value ?

Reply:
Hi @hnhoan , thanks for the elaboration. 

I believe you have a point on PULLUP might not be the correct name in all cases, maybe it should had been LEVEL from the beginning (transistor case). Based on this i believe we could rename all boards to use LEVEL instead of PULLUP, but i suggest we do this isolated after your PR gets in.

When it comes to updating all boards with current code, it should not be needed. The LEDX_PULLUP and LED_PULLUP will be backwards compatible. By default they will use LED_PULLUP, but if you specifically want to override one, LEDX_PULLUP can be used to override the general setting. I'm thinking a bit like this:
https://github.com/I-SYST/micropython/compare/blystnano...glennrub:led_proposal
My main point iin this diff is the mpconfigboard.h. With this code *blueio_tag_evim* can override one LED, while all other board configs are untouched.

Reply:
@glennrub I am fine with that.  Do I need to change anything or you ok to merge?

Reply:
@hnhoan , sorry for the long processing time on my side. 

Content wise i am happy. :) and we follow up with a commit similar to the one i proposed to do some justifications to the names. We can do this in two steps. I propose that we squash the two commits you have here into one when merging. Does this sound OK to you?

Reply:
@glennrub That's fine with me, thanks.